### PR TITLE
align the behavior between REPL, REPL seeded with a script and script

### PR DESCRIPTION
**Pull request by filipw**
We support running a script + dropping you into the REPL using `dotnet script foo.csx -i` start up arguments.
This PR aligns how the output (errors, return values) are dumped into the console when running a REPL with seed vs just running a script.

It also ensures that in such "seed" cases, the arguments are correctly surfaced to the script.

### CODE REVIEW

**Comment by seesharper** [comment]
@filipw Left a comment here, but maybe it should be looked at in another PR ?

**Comment by filipw** [comment]
which comment though? 😁

File: src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs [hunk 1]
```diff
@@ -129,6 +108,9 @@ public virtual void Exit()
 
         private async Task RunFirstScript(ScriptContext scriptContext)
         {
+            foreach (var arg in scriptContext.Args)
+                _globals.Args.Add(arg);
+
             var compilationContext = ScriptCompiler.CreateCompilationContext<object, InteractiveScriptGlobals>(scriptContext);
             _scriptState = await compilationContext.Script.RunAsync(_globals, ex => true).ConfigureAwait(false);
             _scriptOptions = compilationContext.ScriptOptions;
```
[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: Forgot to submit the review. That's why you did not see the comment :)

[the remark this answers is not in the document]
Comment (filipw) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: this is all handled when going through `CreateCompilationContext`, isn't it? here https://github.com/filipw/dotnet-script/blob/9045fe6745313ad412d2d54c3c3d0f3e093e25eb/src/Dotnet.Script.Core/ScriptCompiler.cs#L95

it uses the same `ScriptCompiler` as regular script runner would use

[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: Ah, right. You are probably correct. I'm thinking maybe adding two new tests that verifies this.

one with `#r "nuget:package,1.2.3` as the first line and one with `#load "nuget:scriptpackage, 1.2.2"` and the first line. Just to be sure 😄

[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: @filipw  Something like 

```
 [Fact]
        public async Task NugetPackageReferenceAsTheFirstLine()
        {
            var commands = new[]
            {
                @"#r ""nuget: Automapper, 6.1.1""",
                "using AutoMapper;",
                "typeof(MapperConfiguration)",
                "#exit"
            };

            var ctx = GetRunner(commands);
            await ctx.Runner.RunLoop(false);

            var result = ctx.Console.Out.ToString();
            Assert.Contains("[AutoMapper.MapperConfiguration]", result);
        }
```

[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: And 

```
 [Fact]
        public async Task ScriptPackageReferenceAsTheFirstLine()
        {            
            var commands = new[]
            {
                @"#load ""nuget: simple-targets-csx, 6.0.0""",
                "using static SimpleTargets;",
                "typeof(TargetDictionary)",                
                "#exit"
            };

            var ctx = GetRunner(commands);
            await ctx.Runner.RunLoop(false);           
            var result = ctx.Console.Out.ToString();
            Assert.Contains("[Submission#1+SimpleTargets+TargetDictionary]", result);
        }
```

[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: Probably just nitpicking on my part, but I guess a couple of extra tests does not hurt 😄

[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: @filipw I'll add the tests if you are short on time ? 👍

[the remark this answers is not in the document]
Comment (filipw) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: yep, good idea - I will add them

[the remark this answers is not in the document]
Comment (filipw) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: you were right, it didn't actually work 😂 
see this changeset, should be fine now https://github.com/filipw/dotnet-script/pull/187/commits/75f89349af425de65e73accd1c9307d9f5c6a0aa

[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: Awesome 🥇

[the remark this answers is not in the document]
Comment (seesharper) [reply src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs]: I'd say we are good to go for 0.16.0

**Pull request merged**